libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -47,7 +47,7 @@ bool ReferencesResolver::visit(Identifier const& _identifier)
 {
 	auto declarations = m_resolver.nameFromCurrentScope(_identifier.name());
 	if (declarations.empty())
-		fatalDeclarationError(_identifier.location(), "Undeclared identifier.");
+		declarationError(_identifier.location(), "Undeclared identifier.");
```

Comment:
Does this stop the compiler after the reference resolution stage? Some later stages assume `referencedDeclaration` to be non-null, I think.

Reply:
Yes, it should stop right [here](https://github.com/ethereum/solidity/blob/8d3cfa8cff85b1aed7c1b77a0886a58b09144a36/libsolidity/interface/CompilerStack.cpp#L190).